ocp_resources/utils.py

```diff
@@ -26,30 +27,97 @@ class TimeoutSampler:
     Yielding the output allows you to handle every value as you wish.
 
     Feel free to set the instance variables.
+
+    exceptions_dict should be in the following format:
+    {
+        exception0: [exception0_msg0],
+        exception1: [
+            exception1_msg0,
+            exception1_msg1
+        ],
+        exception2: []
+    }
+
+    Args:
+        wait_timeout (int): Time in seconds to wait for func to return a value equating to True
+        sleep (int): Time in seconds between calls to func
+        func (function): to be wrapped by TimeoutSampler
+        exceptions (tuple): Deprecated: Tuple containing all retry exceptions to pass to TimeoutSampler
+        exceptions_msg (str): Deprecated: String to match exception against
+        exceptions_dict (dict): Definition for exception handling
+        print_log (bool): Print elapsed time to log
     """
 
     def __init__(
         self,
         wait_timeout,
         sleep,
         func,
-        exceptions=None,
-        exceptions_msg=None,
+        exceptions=None,  # Deprecated
+        exceptions_msg=None,  # Deprecated
+        exceptions_dict=None,
         print_log=True,
+        *func_args,
         **func_kwargs,
     ):
         self.wait_timeout = wait_timeout
         self.sleep = sleep
         self.func = func
+        self.func_args = func_args
```

Comment:
Please rebase, func_args was removed (was not used and func_kwargs should be used if needed)

Reply:
corrected